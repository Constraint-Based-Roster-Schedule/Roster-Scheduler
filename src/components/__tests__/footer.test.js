import { render, screen } from "@testing-library/react";
import { getByTestId } from "@testing-library/react";
import Footer from "../footer";
import ReactDom from "react-dom";
import { Button } from "react-bootstrap";
import renderer from "react-test-renderer";
import shallow from "react-test-renderer/shallow"; // ES6
import ConsultantDashboard from "../consultantDashboard";
import { FiberPinSharp, FortTwoTone } from "@mui/icons-material";
test("footer testing", () => {
  render(<Footer />);
  
});
it('matches snapshot',()=>{
  const tree=renderer.create(<Footer></Footer>).toJSON();
  expect(tree).toMatchSnapshot();

})

describe("Test footer component", () => {
    const createInstance = () => {
      render(<Footer/>);
    };
  
    it("render footer component", () => {
      createInstance();
      const footer_detail = screen.getByText("Gallery");
      expect(footer_detail).not.toBeNull();
    });
  });