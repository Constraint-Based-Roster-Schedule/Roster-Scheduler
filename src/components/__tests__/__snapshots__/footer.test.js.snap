// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="MuiBox-root css-kqi55e"
>
  <div
    className="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-spacing-xs-2 css-5yghch-MuiGrid2-root"
  >
    <div
      className="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-5 MuiGrid2-grid-lg-3 css-12cd7m9-MuiGrid2-root"
    >
      <div
        className="img-class"
      >
        <img
          alt=""
          src="logo.png"
        />
      </div>
    </div>
    <div
      className="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 MuiGrid2-grid-md-7 MuiGrid2-grid-lg-8 MuiGrid2-spacing-xs-4 css-wqqmfr-MuiGrid2-root"
    >
      <div
        className="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-lg-3 css-7n4qqf-MuiGrid2-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ppo8o6-MuiPaper-root"
        >
          <div
            className="MuiBox-root css-17regzl"
            id="category-a"
          >
            Follow us
          </div>
          <ul
            aria-labelledby="category-a"
            className="MuiBox-root css-4r15ta"
          >
            <li>
              <td>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="FacebookIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
                  />
                </svg>
              </td>
              <td>
                Facebook
              </td>
            </li>
            <li>
              <td>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="LinkedInIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 3a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2h14m-.5 15.5v-5.3a3.26 3.26 0 0 0-3.26-3.26c-.85 0-1.84.52-2.32 1.3v-1.11h-2.79v8.37h2.79v-4.93c0-.77.62-1.4 1.39-1.4a1.4 1.4 0 0 1 1.4 1.4v4.93h2.79M6.88 8.56a1.68 1.68 0 0 0 1.68-1.68c0-.93-.75-1.69-1.68-1.69a1.69 1.69 0 0 0-1.69 1.69c0 .93.76 1.68 1.69 1.68m1.39 9.94v-8.37H5.5v8.37h2.77z"
                  />
                </svg>
              </td>
              <td>
                LinkedIn
              </td>
            </li>
            <li>
              <td>
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="TwitterIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M22.46 6c-.77.35-1.6.58-2.46.69.88-.53 1.56-1.37 1.88-2.38-.83.5-1.75.85-2.72 1.05C18.37 4.5 17.26 4 16 4c-2.35 0-4.27 1.92-4.27 4.29 0 .34.04.67.11.98C8.28 9.09 5.11 7.38 3 4.79c-.37.63-.58 1.37-.58 2.15 0 1.49.75 2.81 1.91 3.56-.71 0-1.37-.2-1.95-.5v.03c0 2.08 1.48 3.82 3.44 4.21a4.22 4.22 0 0 1-1.93.07 4.28 4.28 0 0 0 4 2.98 8.521 8.521 0 0 1-5.33 1.84c-.34 0-.68-.02-1.02-.06C3.44 20.29 5.7 21 8.12 21 16 21 20.33 14.46 20.33 8.79c0-.19 0-.37-.01-.56.84-.6 1.56-1.36 2.14-2.23z"
                  />
                </svg>
              </td>
              <td>
                Twitter
              </td>
            </li>
          </ul>
        </div>
      </div>
      <div
        className="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-lg-3 css-7n4qqf-MuiGrid2-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ppo8o6-MuiPaper-root"
        >
          <div
            className="MuiBox-root css-9flgt5"
            id="category-b"
          >
            Your Account
          </div>
          <ul
            aria-labelledby="category-b"
            className="MuiBox-root css-n085mf"
          >
            <li>
              Signup
            </li>
            <li>
              Login
            </li>
            <li>
              Home
            </li>
          </ul>
        </div>
      </div>
      <div
        className="MuiGrid2-root MuiGrid2-direction-xs-row MuiGrid2-grid-xs-6 MuiGrid2-grid-lg-3 css-7n4qqf-MuiGrid2-root"
      >
        <div
          className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1ppo8o6-MuiPaper-root"
        >
          <div
            className="MuiBox-root css-9flgt5"
            id="category-c"
          >
            Visit
          </div>
          <ul
            aria-labelledby="category-c"
            className="MuiBox-root css-n085mf"
          >
            <li>
              Contact us
            </li>
            <li>
              About us
            </li>
            <li>
              Gallery
            </li>
          </ul>
        </div>
      </div>
    </div>
    <div
      className="MuiGrid2-root MuiGrid2-container MuiGrid2-direction-xs-row MuiGrid2-grid-xs-12 css-uxq1a1-MuiGrid2-root"
    >
      <div
        className="MuiGrid2-root MuiGrid2-direction-xs-row css-1htzj5z-MuiGrid2-root"
      >
        Â© Copyright TechNext
      </div>
    </div>
  </div>
</div>
`;
